pkg/storage/system_intake_contact.go

```diff
@@ -0,0 +1,81 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+
+	"github.com/google/uuid"
+
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/apperrors"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// CreateSystemIntakeContact creates a new system intake contact object in the database
+func (s *Store) CreateSystemIntakeContact(ctx context.Context, systemIntakeContact *models.SystemIntakeContact) (*models.SystemIntakeContact, error) {
+	euaUserID := appcontext.Principal(ctx).ID()
+	createAt := s.clock.Now().UTC()
+
+	systemIntakeContact.CreatedAt = &createAt
```

Comment:
Both this assignment and the `systemIntakeContact.EUAUserID` assignment should probably be guarded by checking if the field is `nil` first, and only assigning if it's `nil`. That would allow a caller to set their own value for EUA ID/createdAt (such as adding a contact that's _not_ the current user), as well as being consistent with other `Store` methods.

Reply:
@DylanSpOddball I made changes so that the created at can be nil, and removed the initialization of EUA ID based on the current user. The EUA should always be given as part of the input when creating a contact, as it cannot be nil, and it would not make sense to default it to the current user.